sycl/unittests/Extensions/VirtualMemory/Exceptions.cpp

```diff
@@ -0,0 +1,172 @@
+#include <sycl/sycl.hpp>
+
+#include <sycl/ext/oneapi/virtual_mem/physical_mem.hpp>
+#include <sycl/ext/oneapi/virtual_mem/virtual_mem.hpp>
+
+#include "ur_api.h"
+
+#include <helpers/UrMock.hpp>
+
+#include <gtest/gtest.h>
+
+#include <array>
+
+namespace syclext = sycl::ext::oneapi::experimental;
+
+constexpr size_t NumberOfDevices = 3;
+
+std::array<ur_device_handle_t, NumberOfDevices> GlobalDevicesHandle{
+    mock::createDummyHandle<ur_device_handle_t>(),
+    mock::createDummyHandle<ur_device_handle_t>(),
+    mock::createDummyHandle<ur_device_handle_t>(),
+};
+
+ur_result_t setup_urDeviceGet(void *pParams) {
+  auto params = *static_cast<ur_device_get_params_t *>(pParams);
+  if (*params.ppNumDevices)
+    **params.ppNumDevices = 3;
+  if (*params.pphDevices) {
+    if (*params.pNumEntries > 0)
+      (*params.pphDevices)[0] = GlobalDevicesHandle[0];
+    if (*params.pNumEntries > 1)
+      (*params.pphDevices)[1] = GlobalDevicesHandle[1];
+    if (*params.pNumEntries > 2)
+      (*params.pphDevices)[2] = GlobalDevicesHandle[2];
+  }
+  return UR_RESULT_SUCCESS;
+}
+
+template <bool VirtualMemSupported>
+ur_result_t after_urDeviceGetInfo_AllDevices(void *pParams) {
+  auto params = reinterpret_cast<ur_device_get_info_params_t *>(pParams);
+  switch (*params->ppropName) {
+  case UR_DEVICE_INFO_VIRTUAL_MEMORY_SUPPORT: {
```

Comment:
Can we just have early returns instead of switches everywhere?

Reply:
Sure, can be done